src/odemis/driver/xt_client.py

```diff
@@ -2327,8 +2327,14 @@ def _moveTo(self, future: futures.Future, pos: Dict[str, float], rel: bool = Fal
                 self.parent.move_stage(pos, rel=rel)
                 time.sleep(0.1)  # It takes a little while before the stage is being reported as moving
 
+                # Get the target position in absolute coordinates
+                if rel:
+                    target_pos = {ax: val + pos[ax] for ax, val in orig_pos.items()}
+                else:
+                    target_pos = {ax: pos[ax] for ax, val in orig_pos.items()}
+
                 # Drop axes from the original position, which are not important because they have not moved
-                orig_pos = {a: orig_pos[a] for a, nv in pos.items() if nv != orig_pos[a]}
+                orig_pos = {a: orig_pos[a] for a, nv in target_pos.items() if nv != orig_pos[a]}
```

Comment:
Do you really need this? At line 2327 one is already requesting a move to `pos` stating its `rel`. Maybe the `time.sleep(0.1)` is not enough in some cases?

Reply:
We need orig position a bit later when we start checking if the stage have moved to the requested position. Does that answer your question?